spec/researches/keywordCountSpec.js

```diff
@@ -2,41 +2,108 @@
-		expect( keywordCount( mockPaper  ).count ).toBe( 1 );
-		mockPaper = new Paper( "a string of text with the kapaklı in it, density should be 7.7%", { keyword: "kapaklı" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 1 );
-		mockPaper = new Paper( "a string of text with the key-word in it, density should be 7.7%", { keyword: "key-word" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 1 );
-		mockPaper = new Paper( "a string of text with the key_word in it, density should be 7.7%", { keyword: "key_word" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 1 );
-		mockPaper = new Paper( "a string of text with the key_word in it, density should be 0.0%", { keyword: "key word" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 0 );
-		mockPaper = new Paper( "a string of text with the key-word in it, density should be 0.0%", { keyword: "key word" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 0 );
-		mockPaper = new Paper( "a string of text with the key&word in it, density should be 7.7%", { keyword: "key&word" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 1 );
-		mockPaper = new Paper( "<img src='http://image.com/image.png'>", { keyword: "key&word" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 0 );
-		// Consecutive keywords are skipped, so this will match 2 times.
-		mockPaper = new Paper( "This is a nice string with a keyword keyword keyword.", { keyword: "keyword" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 2 );
-		mockPaper = new Paper( "a string of text with the $keyword in it, density should be 7.7%", { keyword: "$keyword" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 1 );
-		mockPaper = new Paper( "a string of text with the Keyword in it, density should be 7.7%", { keyword: "keyword" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 1 );
-		mockPaper = new Paper( "a string of text with the Key word in it, density should be 7.14%", { keyword: "key word" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 1 );
-		mockPaper = new Paper( "a string with quotes to match the key'word, even if the quotes differ", { keyword: "key’word" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 1 );
+	it( "counts a string of text with a keyword in it.", function() {
+		const mockPaper = new Paper( "a string of text with the keyword in it, density should be 7.7%" );
+		expect( keywordCount( mockPaper, mockResearcher ).count ).toBe( 1 );
+	} );
+
+	it( "counts a string of text with no keyword in it.", function() {
+		const mockPaper = new Paper( "a string of text, density should be 0.0%" );
+		expect( keywordCount( mockPaper, mockResearcher ).count ).toBe( 0 );
+	} );
+
+	it( "counts a string of text with multiple occurrences of the keyword in it.", function() {
+		const mockPaper = new Paper( "a string of text with the key word in it, with more key words." );
+		expect( keywordCount( mockPaper, mockResearcherKeyWord ).count ).toBe( 2 );
+	} );
```

Comment:
Please add a test with keyphrase terms being found different number of times. E.g., for keyphrase `key word` and text `a string with three keys (key and another key) and one word` the keyword count should be 1.